lib/ios/RNNComponentPresenter.m

```diff
@@ -43,37 +43,37 @@ - (void)componentDidDisappear {
+    [super applyOptionsOnWillMoveToParentViewController:options];
 }
 
 - (void)applyOptions:(RNNNavigationOptions *)options {
-	[super applyOptions:options];
-	
-	UIViewController* viewController = self.boundViewController;
-	RNNNavigationOptions *withDefault = [options withDefault:[self defaultOptions]];
-	[viewController setBackgroundImage:[withDefault.backgroundImage getWithDefaultValue:nil]];
-	[viewController setTopBarPrefersLargeTitle:[withDefault.topBar.largeTitle.visible getWithDefaultValue:NO]];
-	[viewController setTabBarItemBadgeColor:[withDefault.bottomTab.badgeColor getWithDefaultValue:nil]];
-	[viewController setStatusBarBlur:[withDefault.statusBar.blur getWithDefaultValue:NO]];
-	[viewController setStatusBarStyle:[withDefault.statusBar.style getWithDefaultValue:@"default"] animated:[withDefault.statusBar.animate getWithDefaultValue:YES]];
-	[viewController setBackButtonVisible:[withDefault.topBar.backButton.visible getWithDefaultValue:YES]];
-	[viewController setInterceptTouchOutside:[withDefault.overlay.interceptTouchOutside getWithDefaultValue:YES]];
+    [super applyOptions:options];
+    
+    UIViewController* viewController = self.boundViewController;
+    RNNNavigationOptions *withDefault = [options withDefault:[self defaultOptions]];
+    [viewController setBackgroundImage:[withDefault.backgroundImage getWithDefaultValue:nil]];
+    [viewController setTopBarPrefersLargeTitle:[withDefault.topBar.largeTitle.visible getWithDefaultValue:NO]];
+    [viewController setTabBarItemBadgeColor:[withDefault.bottomTab.badgeColor getWithDefaultValue:nil]];
+    [viewController setStatusBarBlur:[withDefault.statusBar.blur getWithDefaultValue:NO]];
+    [viewController setStatusBarStyle:[withDefault.statusBar.style getWithDefaultValue:@"default"] animated:[withDefault.statusBar.animate getWithDefaultValue:YES]];
+    [viewController setBackButtonVisible:[withDefault.topBar.backButton.visible getWithDefaultValue:YES]];
+    [viewController setInterceptTouchOutside:[withDefault.overlay.interceptTouchOutside getWithDefaultValue:YES]];
     
     if (@available(iOS 13.0, *)) {
         [viewController setBackgroundColor:[withDefault.layout.componentBackgroundColor getWithDefaultValue:UIColor.systemBackgroundColor]];
     } else {
         [viewController setBackgroundColor:[withDefault.layout.componentBackgroundColor getWithDefaultValue:viewController.view.backgroundColor]];
     }
     
-	if (withDefault.topBar.searchBar.hasValue) {
-		BOOL hideNavBarOnFocusSearchBar = YES;
-		if (withDefault.topBar.hideNavBarOnFocusSearchBar.hasValue) {
-			hideNavBarOnFocusSearchBar = withDefault.topBar.hideNavBarOnFocusSearchBar.get;
-		}
-		[viewController setSearchBarWithPlaceholder:[withDefault.topBar.searchBarPlaceholder getWithDefaultValue:@""] hideNavBarOnFocusSearchBar:hideNavBarOnFocusSearchBar];
-	}
-	
-	[_topBarTitlePresenter applyOptions:withDefault.topBar];
+    if (withDefault.topBar.searchBar.hasValue && withDefault.topBar.searchBar.get) {
```

Comment:
Is it safe to check it this way? Is there a better way to make sure `topBar.searchBar` option was provided and it's indeed `true`?